ariadne/cardano/src/Ariadne/Cardano/Knit.hs

```diff
@@ -237,7 +238,8 @@ adaMultiplier = scientific 1 adaExponent
 
 adaToCoin :: Scientific -> Maybe Scientific
 adaToCoin (normalize -> normalized) =
-  if base10Exponent normalized > adaExponent
+  -- To disallow values which is more precisely than 10^-6
+  if base10Exponent normalized < -adaExponent
```

Comment:
Please add a regression test for this fix.

Reply:
How complicated should this test be?

Reply:
The tests you wrote are ok, but I'd also add some unit tests for concrete values (e. g. for the numbers which manifested the bug). Also please check my comments about those tests.